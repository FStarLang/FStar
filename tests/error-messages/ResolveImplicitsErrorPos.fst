module ResolveImplicitsErrorPos

open FStar.Tactics

let tag : unit = ()

[@@ resolve_implicits; tag]
let tau () : Tac unit =
  fail "oops"

let f (#[@@@tag] x:int) : int = x

[@@expect_failure]
let test () =
  let x = f in
  x + 1
