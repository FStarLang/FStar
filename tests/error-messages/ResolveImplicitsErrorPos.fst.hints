[ "†¯î1zý_\u0001]‚¦€Ïzá½", [] ]