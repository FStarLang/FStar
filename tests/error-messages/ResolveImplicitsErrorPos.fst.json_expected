proof-state: State dump @ depth 0 (at the time of failure):
Location: ResolveImplicitsErrorPos.fst(14,13-16,7)
Goal 1/1 (Instantiating meta argument):
(_: Prims.unit) |- _ : Prims.int

>> Got issues: [
{"msg":["Tactic failed","oops"],"level":"Error","range":{"def":{"file_name":"ResolveImplicitsErrorPos.fst","start_pos":{"line":15,"col":10},"end_pos":{"line":15,"col":11}},"use":{"file_name":"ResolveImplicitsErrorPos.fst","start_pos":{"line":15,"col":10},"end_pos":{"line":15,"col":11}}},"number":228,"ctx":["While solving implicits with a tactic","While solving deferred constraints","While typechecking the top-level declaration `let test`","While typechecking the top-level declaration `[@@expect_failure] let test`"]}
>>]
Verified module: ResolveImplicitsErrorPos
All verification conditions discharged successfully
