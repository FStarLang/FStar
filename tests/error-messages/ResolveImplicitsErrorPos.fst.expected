proof-state: State dump @ depth 0 (at the time of failure):
Location: ResolveImplicitsErrorPos.fst(15,2-16,7)
Goal 1/1 (Instantiation of meta argument):
(_: Prims.unit) |- _ : Prims.int

>> Got issues: [
ResolveImplicitsErrorPos.fst(15,10-15,11): (Error 228) user tactic failed: `oops`
>>]
Verified module: ResolveImplicitsErrorPos
All verification conditions discharged successfully
